src/database/settings.ts

```diff
@@ -0,0 +1,52 @@
+export interface DatabaseSetting {
+  /**
+   * The path of the flag to set.
+   */
+  path: string;
+
+  /**
+   * The description of the command.
+   * It should include necessary paddings to be consistent (6 spaces per new line).
+   */
+  description: string;
+
+  /**
+   * Parse the user input to the value.
+   * @return undefined if the input is invalid
+   */
+  parseInput: (input: string) => any | undefined;
+
+  /**
+   * The message to print out if the input is invalid.
+   */
+  parseInputErrorMessge: string;
+}
+
+const defaultWriteSizeLimit: DatabaseSetting = {
+  path: "defaultWriteSizeLimit",
+  description: `
+      Control the maximum size of write for each request. It can be set to small, medium, large or unlimited.
+      If set to unlimited, any arbitrary write will went through, and you are risking
+      blocking database while this write is being processed. For example, deleting at the root
+      of the database is unrevertible and the database will be unavailable until the delete finished.
+      To avoid block large write, you can set this to small(target=10s), medium(target=30s), large(target=60s).
+      The real time database will esitmate the size of each write and abort the request
+      if it abort the write if the estimate is longer than the target time.
+  `,
+  parseInput: (input: string) => {
+    switch (input) {
+      case "small":
+      case "medium":
+      case "large":
+      case "unlimited":
+        return `"${input}"`;
```

Comment:
I have reservations about this returning the input in quotes. Is that strictly necessary? I see it gets put into the body of a request, but the response here shouldn't  bend based on where the output is going to be used. Returning a string with quotes in it seems like a subtlety that would be overlooked and create confusion later on

Reply:
Agree.
I just realized that if `json: true` in request option did what I want.
That is much more clear that wrapping it myself.